online-inference/tensorizer-isvc/tensorizer_hf_isvc/kserve/kserve_api.py

```diff
@@ -0,0 +1,70 @@
+import logging
+import os
+from typing import Dict
+
+import kserve
+import torch
+from load_model import load_model_based_on_type
+from transformers import AutoTokenizer
+
+MODEL_NAME = "gptj"
+MODEL_LOAD_TYPE = os.getenv("MODEL_LOAD_TYPE")
+
+logging.basicConfig(level=kserve.constants.KSERVE_LOGLEVEL)
+logger = logging.getLogger(MODEL_NAME)
+logger.info(f"Model Name: {MODEL_NAME}")
+
+
+class Model(kserve.Model):
+    def __init__(self, name: str):
+        super().__init__(name)
+        self.name = name
+        self.model = None
+        self.tokenizer = None
+        self.eos = None
+        self.ready = False
+
+    def load(self):
+        logger.info(f"Loading {MODEL_NAME}")
+
+        self.model = load_model_based_on_type(model_load_type=MODEL_LOAD_TYPE)
+
+        self.model.eval()
+        torch.manual_seed(100)
+
+        self.tokenizer = AutoTokenizer.from_pretrained("/mnt/pvc/tokenizer")
+        self.eos = self.tokenizer.eos_token_id
+
+        self.ready = True
+
+    def predict(self, payload: Dict, headers: Dict[str, str] = None) -> Dict:
+        if "text" in payload:
```

Comment:
Might want to assert that `payload` is actually a `Dict` to avoid type errors just in case.